mycpp/visitor.py

```diff
@@ -737,6 +737,18 @@ def __init__(self, types: Dict[Expression, Type]) -> None:
         SimpleVisitor.__init__(self)
         self.types = types
 
+    def _GetType(self, expr: Expression) -> Type:
+        if expr in self.types:
+            return self.types[expr]
+        else:
+            self.report_error(expr, "Statement is unreachable")
```

Comment:
How about "Expression is missing a type: statement may be unreachable"

In case this comes up for some other reason?

---

Also it looks like we have  gone from 28 type errors to 29 ...  I thought it might have been less when I added this


https://op.oilshell.org/uuu/github-jobs/10773/dev-minimal.wwz/_tmp/soil/logs/oils-types.txt

https://op.oilshell.org/uuu/github-jobs/10779/dev-minimal.wwz/_tmp/soil/logs/oils-types.txt

I will add a CI check for this ... I added it to the CI, but it doesn't check right now